pkg/kubelet/dockertools/docker_manager.go

```diff
@@ -1832,23 +1831,67 @@ func addNDotsOption(resolvFilePath string) error {
 		return fmt.Errorf("ResolvConfPath %q does not exist", resolvFilePath)
 	}
 
-	glog.V(4).Infof("DNS ResolvConfPath exists: %s. Will attempt to add ndots option: %s", resolvFilePath, ndotsDNSOption)
+	curResolvFileContent, err := getFileContent(resolvFilePath)
+	if err != nil {
+		glog.Errorf("DNS ResolvConfPath exists: %s. But failed to read its content: %v", resolvFilePath, err)
+	} else {
+		glog.V(4).Infof("DNS ResolvConfPath exists: %s. Its content is following:\n %s", resolvFilePath, curResolvFileContent)
+	}
+
+	resolvFileContent := ""
 
-	if err := appendToFile(resolvFilePath, ndotsDNSOption); err != nil {
-		glog.Errorf("resolv.conf could not be updated: %v", err)
-		return err
+	for _, srv := range dns {
+		if resolvFileContent == "" {
```

Comment:
rather than this test and other newline mangling, why not accumulate a []string (one per line) and then join for the write.